### DISCUSSION TITLE
nuxt3/beanstalk: server node-modules

### DISCUSSION THREAD

**Discussion by rolanday** [nuxt/nuxt#16757] [Questions]
Hello,

I'm new to Nuxt (coming from Express) and seeking guidance on how to handle ./output/server/node-modules on beanstalk. The corresponding package.json has bundledDependencies, and is private. Hence, afaik I need to include the entire node-modules folder contents in my deployment. Is there a better way?

Much appreciate any help!

-Roland.

**Comment by sevillaarvin** [comment]
Hmm, I don't think that's how `bundledDependencies` work. I may be wrong but, I think this is how it works:

If you publish your project, `bundledDependencies` will include the specified dependencies in the published project. That way if the specified dependencies gets deleted from npm for whatever reason, your project still works.

**Comment by pi0** [comment]
Hi @rolanday. Generated `package.json` inside `.output` directory, is only an informative meta-data about used dependencies in production bundle. `node_modueles` inside output is already tree-shaken with only required files of used packages. You can safely deploy `.output` without any package manager installs that readys `.output/package.json`

Alternatively, you can use `nitro.externals.trace: false` from `nuxt.config` to bundle everything but it usually has same effect on `.output` size and sometimes makes troubles.
